From 55c548feed55e79df5d001da25943aa0219b361e Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Wed, 3 Feb 2016 18:39:43 +0200
Subject: [PATCH 0788/1240] fix: comphy: sata: remove rx init check

- The rx init is enabled only after the SATA MAC
  initiates transactions

Change-Id: I8766922f8ad8f7fd048e21080c09588bc9962db3
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27190
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/phy/comphy_cp110.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/phy/comphy_cp110.c b/drivers/phy/comphy_cp110.c
index 801ff74..6f58cf9 100644
--- a/drivers/phy/comphy_cp110.c
+++ b/drivers/phy/comphy_cp110.c
@@ -443,21 +443,18 @@ static int comphy_sata_power_up(u32 lane, void __iomem *hpipe_base, void __iomem
 
 	debug("stage: Check PLL\n");
 	data = readl(sd_ip_addr + SD_EXTERNAL_STATUS0_REG);
+
 	/* check the PLL TX */
 	if ((data & SD_EXTERNAL_STATUS0_PLL_TX_MASK) == 0) {
 		error("SD_EXTERNAL_STATUS0_PLL_TX is 0\n");
 		ret = 0;
 	}
+
 	/* check the PLL RX */
 	if ((data & SD_EXTERNAL_STATUS0_PLL_RX_MASK) == 0) {
 		error("SD_EXTERNAL_STATUS0_PLL_RX is 0\n");
 		ret = 0;
 	}
-	/* check the RX init done */
-	if ((data & SD_EXTERNAL_STATUS0_RX_INIT_MASK) == 0) {
-		error("SD_EXTERNAL_STATUS0_RX_INIT is 0\n");
-		ret = 0;
-	}
 
 	debug_exit();
 	return ret;
-- 
1.9.1

